module/zcommon/zfs_namecheck.c

```diff
@@ -222,7 +222,9 @@ entity_namecheck(const char *path, namecheck_err_t *why, char *what)
 		}
 
 		/* Validate the contents of this component */
+		int component_length = 0;
```

Comment:
This seems to be too much work. Can this logic be replaced by a simpler strncmp()?

Reply:
I do not think so. We need to compare with every component. strncmp would not provide the desired.